Intersect.Client/Entities/Entity.cs

```diff
@@ -1386,6 +1386,147 @@ public virtual void DrawName(Color textColor, Color borderColor = null, Color ba
+                    switch (Type)
+                    {
+                        case -1:
+                            // Default Tag when entity has aggression towards a target.
+                            tagFileName = Options.Npc.AggressiveTagIcon;
+
+                            break;
+                        case 0:
+                            // Default Tag when entity Attacks when attacked.
+                            tagFileName = Options.Npc.AttackWhenAttackedTagIcon;
+
+                            break;
+                        case 1:
+                            // Default Tag when entity Attacks on sight.
+                            tagFileName = Options.Npc.AttackOnSightTagIcon;
+
+                            break;
+                        case 3:
+                            // Default Tag when entity is Guard.
+                            tagFileName = Options.Npc.GuardTagIcon;
+
+                            break;
+                        case 2:
+                        default:
+                            // Default Tag when entity is Neutral.
+                            tagFileName = Options.Npc.NeutralTagIcon;
+
+                            break;
+                    }
+                }
+                // Now that we have the name of the sprite file, we load it as a texture.
+                GameTexture tagTexture = Globals.ContentManager.GetTexture(GameContentManager.TextureType.Tag, tagFileName);
+                // Before we draw the sprite, lets have it's position set.
+                switch (npcTagPos)
+                {
+                    case NpcTagPos.Above:
+                    default:
+                        // Position the tag 2 pixels above the name label.
+                        x = nameCentHorPos - (tagTexture.GetWidth() / 2);
+                        y = nameVertPos - tagTexture.GetHeight() - 2;
+
+                        break;
+                    case NpcTagPos.Under:
+                        // Position the tag 2 pixels under the name label.
+                        x = nameCentHorPos - (tagTexture.GetWidth() / 2);
+                        y = nameVertPos + nameSize.Y + 2;
+
+                        break;
+                    case NpcTagPos.Prefix:
+                        // Position the tag as prefix (2 pixels left from the name label).
+                        x = nameCentHorPos - (nameSize.X / 2) - tagTexture.GetWidth() - 6;
+                        y = nameVertPos + (nameSize.Y / 2) - (tagTexture.GetHeight() / 2);
+
+                        break;
+                    case NpcTagPos.Suffix:
+                        // Position the tag as suffix (2 pixels right from the name label).
+                        x = nameCentHorPos + (nameSize.X / 2) + 6;
+                        y = nameVertPos + (nameSize.Y / 2) - (tagTexture.GetHeight() / 2);
+
+                        break;
+                }
+                // And finally, we draw the tag.
+                if (tagTexture != null)
```

Comment:
The opposite check should happen before the `switch` (if null then return).

If the texture is null there's no reason to do any of the computations, and if the texture is null the switch will actually crash due to null reference exceptions anyway.

Reply:
👍🏽 

```
                // Now that we have the name of the sprite file, we load it as a texture.
                var tagTexture = Globals.ContentManager.GetTexture(GameContentManager.TextureType.Tag, tagFileName);
                // If the texture is null, we do nothing.
                if (tagTexture == null)
                {
                    return;
                }
                // Before we draw the sprite, lets have it's position set.
                switch (tagPos)
                    {
```